process/mqtt.hpp

```diff
@@ -0,0 +1,190 @@
+* modification, are permitted provided that the following conditions
+* are met:
+* 1. Redistributions of source code must retain the above copyright
+*    notice, this list of conditions and the following disclaimer.
+* 2. Redistributions in binary form must reproduce the above copyright
+*    notice, this list of conditions and the following disclaimer in
+*    the documentation and/or other materials provided with the
+*    distribution.
+* 3. Neither the name of the Company nor the names of its contributors
+*    may be used to endorse or promote products derived from this
+*    software without specific prior written permission.
+*/
+#ifndef CACHE_CPP_TPLUGIN_HPP
+#define CACHE_CPP_TPLUGIN_HPP
+
+
+#ifdef WITH_NEMEA
+#include "fields.h"
+#endif
+
+#include <ipfixprobe/process.hpp>
+#include <ipfixprobe/flowifc.hpp>
+#include <ipfixprobe/packet.hpp>
+#include <ipfixprobe/ipfix-elements.hpp>
+#include <ipfixprobe/ipfix-basiclist.hpp>
+#include <ipfixprobe/utils.hpp>
+#include <numeric>
+#include <cstring>
+#include "sstream"
+
+namespace ipxp {
+
+#define MQTT_UNIREC_TEMPLATE "MQTT_TYPE_CUMULATIVE, MQTT_VERSION, MQTT_CONNECTION_FLAGS, MQTT_KEEP_ALIVE, MQTT_CONNECTION_RETURN_CODE, MQTT_PUBLISH_FLAGS, MQTT_TOPICS"
+
+UR_FIELDS (
+    uint16 MQTT_TYPE_CUMULATIVE,
+    uint8 MQTT_VERSION,
+    uint8 MQTT_CONNECTION_FLAGS,
+    uint16 MQTT_KEEP_ALIVE,
+    uint8 MQTT_CONNECTION_RETURN_CODE,
+    uint8 MQTT_PUBLISH_FLAGS,
+    string MQTT_TOPICS
+)
+
+class MQTTOptionsParser : public OptionsParser
+{
+public:
+    uint32_t m_maximal_topic_count; ///< Maximal count of topics from Publish packet header to store for each flow
+
+    MQTTOptionsParser() : OptionsParser("mqtt", "Parse MQTT traffic"), m_maximal_topic_count(0)
+    {
+        register_option(
+            "tc",
+            "topiccount",
+            "count",
+            "Export first tc topics from Publish packet header. Topics are separated by #. Default value is 0.",
+            [this](const char *arg){try {
+                    m_maximal_topic_count = str2num<decltype(m_maximal_topic_count)>(arg);
+                } catch(std::invalid_argument &e) {
+                    return false;
+                }
+                return true;
+            },
+            OptionFlags::RequiredArgument);
+    }
+};
+
+struct RecordExtMQTT : public RecordExt {
+    static int REGISTERED_ID;
+    RecordExtMQTT() :
+        RecordExt(REGISTERED_ID),
+        type_cumulative(0),
+        version(0),
+        connection_flags(0),
+        keep_alive(0),
+        session_present_flag(false),
+        connection_return_code(0),
+        publish_flags(0) {}
+
+    uint16_t type_cumulative; ///< Types of packets presented during communication and session present flag. DISCONNECT(1b) | PINGRESP(1b) | PINGREQ(1b) | UNSUBACK(1b) | UNSUBSCRIBE(1b) | SUBACK(1b) | SUBSCRIBE(1b) | PUBCOMP(1b) | PUBREL(1b) | PUBREC(1b) | PUBACK(1b) | PUBLISH(1b) | CONNACK(1b) | CONNECT(1b) | session present(1b)

```

Comment:
shorten lines that are too long... 328 characters in 1 line is too much (100 per line ideally)